### PR TITLE
client: support getting ceph.dir.rsnaps vxattr

### PR DESCRIPTION
**Pull request by gmayyyha**
Fixes: https://tracker.ceph.com/issues/47168
Signed-off-by: Yanhu Cao <gmayyyha@gmail.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by gmayyyha** [comment]
@jtlayton @batrick pls review.

File: src/mds/CInode.cc [hunk 1]
```diff
@@ -4037,6 +4037,7 @@ int CInode::encode_inodestat(bufferlist& bl, Session *session,
     encode(any_i->change_attr, bl);
     encode(file_i->export_pin, bl);
     encode(snap_btime, bl);
+    encode(file_i->rstat.rsnaps, bl);
```
Comment (batrick) [inline src/mds/CInode.cc:4040]: You need to bump the version in `ENCODE_START` above.

File: src/messages/MClientReply.h [hunk 2]
```diff
@@ -192,6 +192,7 @@ struct InodeStat {
       }
       if (struct_v >= 3) {
         decode(snap_btime, p);
+        decode(rstat.rsnaps, p);
```
Comment (batrick) [inline src/messages/MClientReply.h:195]: ditto for DECODE

Comment (gmayyyha) [reply src/messages/MClientReply.h:197]: @batrick done.

**Comment by gmayyyha** [comment]
jenkins test make check

**Pull request merged**